File: src/1356-Sort-Integers-by-The-Number-of-1-Bits/1356.js
var sortByBits = function(arr) {
    let bin = function(x) {
        let res = 0;
        while (x) {
            x = x & (x - 1);
            res++;
        }
        return res;
    }
    arr.sort((a, b) => (bin(a) * 10000 + a) - (bin(b) * 10000 + b));
    return arr;
};

File: src/1356-Sort-Integers-by-The-Number-of-1-Bits/1356.go
func sortByBits(arr []int) []int {
    sort.Slice(arr, func(i, j int) bool {
        a, b := bin(arr[i]), bin(arr[j])
        if a == b {
            return arr[i] < arr[j]
        }
        return a < b
    })
    return arr
}

func bin(x int) int {
    res := 0
    for x > 0 {
        x = x & (x - 1)
        res++
    }
    return res
}